src/utils/errors.rs

```diff
@@ -0,0 +1,92 @@
+use std::fmt;
+use thiserror::Error;
+use rocket::request::Request;
+use rocket::response::{self, Response, Responder};
+use rocket::http::{ContentType, Status};
+use serde_json;
+use serde::{Serialize, Deserialize};
+use std::io::Cursor;
+use log::debug;
+use anyhow::Result;
+
+pub type ApiResult<T, E = ApiError> = Result<T, E>;
+
+#[derive(Error, Debug, Serialize)]
+pub struct ApiError {
+    pub status: u16,
+    pub message: ApiErrorMessage,
+}
+
+#[derive(Debug, Serialize, PartialEq)]
+#[serde(untagged)]
```

Comment:
Maybe should be tagged? For the clients' easy consumption.

Reply:
They should be serialized all in the same format and that is:
```json
{
    "code": 1,
    "message": "Checksum address validation failed",
    "arguments": [
        "some_address"
    ]
}
```

I would use the polymorphism only internally to make it easier to create the error, but if you parse the backend error I would always use this for rendering out ... as said for error happening in the client gateway I would propose status code "1337" (or something else agreed on with the client teams), and proper message + arguments.

Reply:
So this is still not correctly serialized. What I expect:
- Gateway error:
Status code: 500
Payload:
```json
{
    "code": 1337,
    "message": "Serde serialization error: <serde error message>"
}
``` 
- Well formatted service error:
Status code: <service status code>
Payload: <service error payload>, e.g.:
```json
{
    "code": 1,
    "message": "Checksum address validation failed",
    "arguments": [
        "some_address"
    ]
}
``` 
- Unformatted service error:
Status code: <service status code>
Payload:
```json
{
    "code": 42,
    "message": "Unknown service error: <service response>",
}
``` 

Something along these lines, this way every error returned by the client gateway is of the format:

```json
{
    "code": <error code>,
    "message": "<error message>",
    "arguments": [
        <0..n arguments>
    ]
}
```

We should communicate with the clients that we use new status code and if necessary align them with them (if they prefer something else than `1337` and `42`)